core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -50,12 +50,16 @@ oppia.factory('explorationData', [
     var explorationData = {
       explorationId: explorationId,
       autosaveChangeList: function(changeList, successCallback, errorCallback) {
+        // First save locally to be retrieved later if save is unsuccessful.
+        localSaveService.saveExplorationDraft(explorationId, changeList,
+          draftChangeListId);
         $http.put(explorationDraftAutosaveUrl, {
           change_list: changeList,
           version: explorationData.data.version
         }).then(function(response) {
+          draftChangeListId = response.data.draft_change_list_id;
+          localSaveService.removeExplorationDraft(explorationId);
           if (successCallback) {
-            draftChangeListId = response.data.draft_change_list_id;
             successCallback(response);
           }
         }, function() {
```

Comment:
Yep I was planning to add it on the next PR which will also deal with the case where the user loses changes due to them not having access to localStorage for whatever reason. It wasn't part of the original milestones but like you said I think it's an important use-case. I'd be happy to add it to this PR if you'd prefer of course.